### PR TITLE
Updated pinned versions of prepackaged servers

### LINKED ISSUE #2065: Ensure all model servers have pinned requirements and the full requirements.txt is included in the docs
Currently the requirements on the sklearn model server are not pinned down, resulting in un-predictable "latest" versions for the dependencies available. We'll need to pin down these versions and ensure that all the dependencies are explicitly provided in the documentation. It may be worth adding a mention in the prepackaged server pages linking to the custom server doc page to specify that the image can be overriden with custom images with specific dependencies as required.  Relevant to #2012.

### PR DESCRIPTION
**Pull request by axsaucedo**
Fixes #2065

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **axsaucedo**
You can assign the PR to them by writing `/assign @axsaucedo` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Tue Sep  8 12:48:57 UTC 2020
The logs for [pr-build] [1] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2399/1.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2399 --build=1

**Comment by seldondev** [comment]
Tue Sep  8 12:48:57 UTC 2020
The logs for [integration] [3] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2399/3.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2399 --build=3

**Comment by seldondev** [comment]
Tue Sep  8 12:49:23 UTC 2020
The logs for [lint] [2] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2399/2.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2399 --build=2

**Comment by seldondev** [comment]
@axsaucedo: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 3336172b6ac0e7556f516cf13af0500bf6f2b400 | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Tue Sep  8 14:54:02 UTC 2020
The logs for [integration] [4] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2399/4.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2399 --build=4

**Comment by seldondev** [comment]
@axsaucedo: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 3336172b6ac0e7556f516cf13af0500bf6f2b400 | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Tue Sep  8 15:30:08 UTC 2020
The logs for [pr-build] [5] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2399/5.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2399 --build=5

**Comment by seldondev** [comment]
Tue Sep  8 15:30:11 UTC 2020
The logs for [lint] [6] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2399/6.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2399 --build=6

**Comment by seldondev** [comment]
Tue Sep  8 20:25:44 UTC 2020
The logs for [integration] [7] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2399/7.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2399 --build=7

**Comment by seldondev** [comment]
@axsaucedo: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 0aad072b44142d536fa6fde5eb8569ad5ec6bc93 | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Tue Sep  8 21:02:32 UTC 2020
The logs for [pr-build] [8] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2399/8.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2399 --build=8

**Comment by seldondev** [comment]
Tue Sep  8 21:02:34 UTC 2020
The logs for [lint] [9] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2399/9.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2399 --build=9

**Comment by seldondev** [comment]
Tue Sep  8 21:05:17 UTC 2020
The logs for [integration] [10] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2399/10.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2399 --build=10

**Comment by seldondev** [comment]
@axsaucedo: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 9fa30e8201ac647ad3f7221371e2752e4c0a2531 | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by axsaucedo** [comment]
Seems only failing test is a flaky one (rolling update), should be ready for review

/test integration
/cc @cliveseldon 
/cc @adriangonz

**Comment by seldondev** [comment]
Wed Sep  9 05:52:17 UTC 2020
The logs for [integration] [11] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2399/11.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2399 --build=11

File: doc/source/servers/xgboost.md [hunk 1]
```diff
@@ -4,10 +4,13 @@ If you have a trained XGBoost model saved you can deploy it simply using Seldon'
 
 Prequisites:
 
-  * Use xgboost v0.82
```
Comment (cliveseldon) [inline doc/source/servers/xgboost.md:7]: Might be worth linking to https://xgboost.readthedocs.io/en/latest/tutorials/saving_model.html#a-note-on-backward-compatibility-of-models-and-memory-snapshots

XGBoost is backwards compartible post 1.0.

Not sure why we would stick with 0.82 when latest is 1.2.0 and 1.2.0 can load pre 1.0 models (with a warning).

**Comment by seldondev** [comment]
Thu Sep 10 15:07:28 UTC 2020
The logs for [pr-build] [12] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2399/12.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2399 --build=12

**Comment by seldondev** [comment]
Thu Sep 10 15:07:30 UTC 2020
The logs for [lint] [13] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2399/13.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2399 --build=13

**Comment by seldondev** [comment]
Thu Sep 10 15:09:03 UTC 2020
The logs for [integration] [14] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2399/14.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2399 --build=14

**Comment by seldondev** [comment]
Fri Sep 11 09:47:50 UTC 2020
The logs for [pr-build] [15] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2399/15.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2399 --build=15

**Comment by axsaucedo** [comment]
@cliveseldon I've now updated xgboost to 1.2 - should be good to go

File: doc/source/servers/xgboost.md [hunk 2]
```diff
@@ -4,10 +4,13 @@ If you have a trained XGBoost model saved you can deploy it simply using Seldon'
...
   * The model must be named `model.bst`
   * You must save your model using `bst.save_model(file_path)`
   * The model is loaded with `xgb.Booster(model_file=model_file)`
+  * Dependencies (otherwise it may not work):
+      + scikit-learn == 0.20.3
```
Comment (adriangonz) [inline doc/source/servers/xgboost.md:11]: Do we need `scikit-learn` for the XGBoost server?

File: doc/source/servers/sklearn.md [hunk 3]
```diff
@@ -5,7 +5,10 @@ If you have a trained SKLearn model saved as a pickle you can deploy it simply u
 Pre-requisites:
 
   * The model pickle must be saved using joblib and presently be named `model.joblib`
-  * We presently use sklearn version 0.23.2. Your pickled model must be compatible with this version
+  * Installed dependencies (may not work if versions don't match):
+      + sklearn == 0.23.2
```
Comment (adriangonz) [inline doc/source/servers/sklearn.md:9]: nit: maybe use the full package name (i.e. `scikit-learn`) here?

Comment (axsaucedo) [reply doc/source/servers/sklearn.md:9]: Good catch!

**Comment by seldondev** [comment]
Fri Sep 11 15:50:20 UTC 2020
The logs for [lint] [17] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2399/17.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2399 --build=17

**Comment by seldondev** [comment]
Fri Sep 11 15:50:22 UTC 2020
The logs for [pr-build] [16] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2399/16.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2399 --build=16

**Comment by seldondev** [comment]
Thu Sep 17 09:26:19 UTC 2020
The logs for [pr-build] [18] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2399/18.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2399 --build=18

**Comment by seldondev** [comment]
Thu Sep 17 09:26:22 UTC 2020
The logs for [lint] [19] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2399/19.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2399 --build=19

**Pull request merged**